src/main/scala/firrtl/passes/InferResets.scala

```diff
@@ -0,0 +1,200 @@
+// See LICENSE for license details.
+
+package firrtl.passes
+
+import firrtl._
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.traversals.Foreachers._
+import firrtl.analyses.InstanceGraph
+import firrtl.graph.{DiGraph, MutableDiGraph}
+import firrtl.annotations.{ReferenceTarget, TargetToken}
+import firrtl.Utils.toTarget
+import collection.mutable
+
+import scala.util.{Try, Success, Failure}
+
+/** Infers the concrete type of [[Reset]]s by their connections
+  * This is a global inference because ports can be of type [[Reset]]
```

Comment:
I'd document the behavior of the case if two instances of the same module have inferred different reset ports.

Reply:
done